### ISSUE TITLE
WebdriverIO in watch-mode "forgets" hostname and protocol if not explicitly specified

### ISSUE THREAD

**Issue by flunderpero** [webdriverio/webdriverio#6557]
**Environment (please complete the following information):**
 - **WebdriverIO version:** [7.1.0]
 - **Mode:** [WDIO Testrunner]
 - **If WDIO Testrunner, running sync/async:** [async]

**Config of WebdriverIO**
```
export const config: WebdriverIO.Config = {
    runner: "local",
    specs: ["__specs__/*.spec.js"],
    specFileRetries: 0,
    capabilities: [{browserName: "chrome", "goog:chromeOptions": {args: chrome_args}}],
    port: 9515,
    path: "/",

    services: [
        ["chromedriver", {logFileName: `${test_dir}/chromedriver.log`, port: 9515}],
    ],
}
```

Note: The bug happens with the default service as well.

**Describe the bug**
WebdriverIO works fine until I try to use the watch mode. Then the first run is still ok but the consecutive ones fail with:
```
UnsupportedProtocolError: Unsupported protocol "undefined:"
    at normalizeArguments (/Users/pero/src/pero/cling-main/node_modules/got/dist/source/core/index.js:420:23)
    at normalizeArguments (/Users/pero/src/pero/cling-main/node_modules/got/dist/source/as-promise/core.js:34:31)
    at Object.got [as default] (/Users/pero/src/pero/cling-main/node_modules/got/dist/source/create.js:88:39)
    at WebDriverRequest._request (/Users/pero/src/pero/cling-main/node_modules/webdriver/build/request.js:126:41)
    at WebDriverRequest.makeRequest (/Users/pero/src/pero/cling-main/node_modules/webdriver/build/request.js:68:21)
    at Browser.protocolCommand (/Users/pero/src/pero/cling-main/node_modules/webdriver/build/command.js:74:24)
    at Browser.wrapCommandFn (/Users/pero/src/pero/cling-main/node_modules/@wdio/utils/build/shim.js:79:38)
```

It works once I explicitly configure hostname and protocol:
```
config = {
...
    hostname: "localhost",
    protocol: "http",
...
}

**Comment by jbblanchet** [comment]
We have the same problem, and the workaround recommended here worked with one caveat: once we added the hostname and protocol, the tests started failing with `Invalid URL: http://localhost:undefined/session/9f5cccf8ded410a398f94c93e16d4292/window/rect`. This forced us to set the port explicitly.

**Comment by flunderpero** [comment]
@jbblanchet I guess you have to set `port`, too. Could this be?

**Comment by jbblanchet** [comment]
@flunderpero Yes, this is what we've done and it fixed it, like I mentioned at the end. Sorry if that was unclear. Had to dig around a bit to understand that this was the webdriver port, and not the website port, as both the hostname and protocol were the website.

**Comment by flunderpero** [comment]
@jbblanchet Ah, sorry. I was in the middle of something else and just wanted to help out quickly. Great to hear that it worked in the end.

**Comment by christian-bromann** [comment]
@flunderpero I can't reproduce this. What I've done is:

```
$ mkdir issue6557
$ cd issue6557
$ npm i @wdio/cli
$ npx wdio config --yes
```
Then I changed the service list to:
```js
    services: [["chromedriver", {logFileName: `${__dirname}/chromedriver.log`, port: 9500}]],
```
and added
```js
    port: 9500,
    path: "/",
```
to the config. It all passes fine. Can you provide a reproducible example?

**Comment by christian-bromann** [comment]
I created a PR with a fix for it. Let's continue our convo there.

**Comment by jbblanchet** [comment]
@christian-bromann the fix didn't actually fix the issue for us. I've added a repro repo in https://github.com/jbblanchet/webdriver-bug, which is pretty vanilla. Please let me know if you need more details, or if you prefer I open a separate bug.

**Comment by jbblanchet** [comment]
@christian-bromann just saw the tags you added. I don't mind trying picking this up, but I just left on vacation for a week, and I have to onboard a new dev when I return, so it could be 10-15 days before I'm able to contribute.

**Comment by christian-bromann** [comment]
Fixed in #6759
